### PR TITLE
Rename TorchDynamo -> Dyanamo in the dynamo tutorial doc

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123431

Less verbose and it aligns it with the dynamo deepdive

### CODE REVIEW

File: docs/source/torch.compiler_dynamo_overview.rst [hunk 1]
```diff
@@ -1,10 +1,10 @@
-TorchDynamo Overview
-====================
+Dynamo Overview
+===============
 
 Before you read this section, read :ref:`torch.compiler_overview`.
 
-**TorchDynamo** is a Python-level Just-In-Time (JIT) compiler designed to make
-unmodified PyTorch programs faster. TorchDynamo hooks into the frame evaluation
+**Dynamo** is a Python-level Just-In-Time (JIT) compiler designed to make
```
Comment (peterbell10) [inline docs/source/torch.compiler_dynamo_overview.rst:6]: Perhaps copy the deepdive which begins with 

> TorchDynamo (or simply Dynamo)

**Comment by lezcano** [comment]
PTAL

**Review by peterbell10 — APPROVED** [review]

**Pull request closed without merging**